From 18137474bcc917bcd9cfd23c2363cc072190cf17 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:20 -0400
Subject: [ethernet] cxgb4: Decode PCIe Gen3 link speed

Message-id: <1410475447-94392-62-git-send-email-svemuri@redhat.com>
Patchwork-id: 93854
O-Subject: [RHEL7.1 PATCH BZ 1124947 061/108] cxgb4: Decode PCIe Gen3 link speed
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Add handling for " 8 GT/s" in print_port_info().

Signed-off-by: Roland Dreier <roland@purestorage.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit d2e752db6d05374a35dddb2e17864fe310fbcf69)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index 11e0792..b04ed15 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -5872,6 +5872,8 @@ static void print_port_info(const struct net_device *dev)
   spd = " 2.5 GT/s";
  else if (adap->params.pci.speed == PCI_EXP_LNKSTA_CLS_5_0GB)
   spd = " 5 GT/s";
+ else if (adap->params.pci.speed == PCI_EXP_LNKSTA_CLS_8_0GB)
+  spd = " 8 GT/s";
 
  if (pi->link_cfg.supported & FW_PORT_CAP_SPEED_100M)
   bufp += sprintf(bufp, "100/");
-- 
1.7.1